### PR TITLE
Azure Advisor Cost Optimization Workbook

### PR DESCRIPTION
**Pull request by arthurclares**
## PR Checklist

* [x ] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
Cost Optimization workbook initial release.
* [ x] Validate your changes using one or more of the [testing](../Documentation/Testing.md) methods.
Tested in the local repo.

### If adding or updating templates:
* [ ] post a screenshot of templates and/or gallery changes
* [ ] ensure your template has a corresponding gallery entry in the gallery folder
* [ ] If you are adding a new template, add your team and template/gallery file(s) to the CODEOWNERS file. CODEOWNERS entries should be teams, not individuals
* [ ] ensure all steps have meaningful names
* [ ] ensure all parameters and grid columns have display names set so they can be localized
* [ x] ensure that parameters id values are unique __or they will fail PR validation__ (parameter ids are used for localization)
* [x ] ensure that steps names are unique __or they will fail PR validation__ (step names are used for localization)
* [x ] grep `/subscription/` and ensure that your parameters don't have any hardcoded resourceIds __or they will fail PR validation__
* [ x] remove `fallbackResourceIds` and `fromTemplateId` fields from your template workbook __or they will fail PR validation__

### CODE REVIEW

File: gallery/workbook/Azure Advisor.json [hunk 1]
```diff
@@ -17,6 +17,13 @@
 					"author": "Microsoft",
 					"name": "Service Retirement (Preview)",
 					"description": "Track service retirements for your Azure resources"
+				},
+				{
+					"id": "Workbooks/Azure Advisor/Cost Optimization",
+					"author": "Microsoft",
+					"name": "Cost Optimization (Preview)",
+					"description": "This workbook surfaces key recommendations that can directly impact cost savings. These recommendations include identifying and addressing idle resources, leveraging Azure Advisor recommendations such as Reserved Instances and Savings Plans for Compute, and optimizing the usage of oversized VMs. ",
```
Comment (pkrishnanms) [inline gallery/workbook/Azure Advisor.json:25]: can the description be shortened, only a few words would be visible in the UI

File: Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook [hunk 2]
```diff
@@ -0,0 +1,4696 @@
...
...Ps|\r\n|v1.1.8| 16/02/2022 | -Added untagged resources list <br> -Added storage account tiers <br> -Added a link to storage account insights for each legacy storage account <br> -Added AHUB check for windows VMSS <br> -Fixed unattached disk query <br> -Fixed App service query (not showing all apps in a service plan) <br> -Added \"Cost Information\" tab|\r\n|v1.1.9| 14/06/2022 | - Fixed \"unattached disk\" query in the cost information tab |\r\n|v1.2.0| 24/07/2022 | - Removed FileStorage account type from V2 query <br> - Added new column in \"Cost Information\" tab to show total VM cost with AHUB <br> - Changed VPN metrics aggregation <br> - Changed column some names |\r\n|v1.2.1| 08/11/2022 | - Add Front Door metrics <br> - Changed cost information billing for VMs without AHUB to TheLastBillingMonth <br> - Increased row limit to in compute queries  <br> - Updated AKS query to show Spot VMs  <br> - Fixed Linux AHUB query to show RHEL license <br> - Add subscriptions advisor recommendations |\r\n|v1.2.2| December/2022 | - Minor fix to the Storage Advisor query |\r\n|v1.3.0 | March /2023 | - Fixed Cost Information Tab |\r\n|v2.0.0 | March /2023 | - Added Azure Hybrid Benefit tab|\r\n"
```
Comment (pkrishnanms) [inline Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook:7]: is the Change Log component needed

File: Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook [hunk 3]
```diff
@@ -0,0 +1,4696 @@
...
+            "type": 12,
+            "content": {
+              "version": "NotebookGroup/1.0",
+              "groupType": "editable",
+              "items": [
+                {
+                  "type": 3,
+                  "content": {
+                    "version": "KqlItem/1.0",
+                    "query": "advisorresources\r\n| where type == \"microsoft.advisor/recommendations\"\r\n| where tostring (properties.category) has \"Cost\"\r\n| where properties.impactedField has \"Network\"\r\n| project AffectedResource=tostring(properties.resourceMetadata.resourceId),Impact=properties.impact,resourceGroup,AdditionaInfo=properties.extendedProperties,subscriptionId,Recommendation=tostring(properties.shortDescription.problem),name\r\n| where resourceGroup in ({ResourceGroup})",
```
Comment (pkrishnanms) [inline Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook:250]: I assume all the text and column names would be updated. Would that be done in this PR or only later?

File: Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook (hunk 2 above)
Comment (arthurclares) [reply Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook:7]: That is a good point, we will remove it.

FYI @sebassem

File: gallery/workbook/Azure Advisor.json (hunk 1 above)
Comment (arthurclares) [reply gallery/workbook/Azure Advisor.json]: shortened.

File: Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook [hunk 4]
```diff
@@ -0,0 +1,4696 @@
...
+          {
+            "id": "ca40468d-4518-43bf-ac6e-0a11d7331e12",
+            "cellValue": "SelectedTab",
+            "linkTarget": "parameter",
+            "linkLabel": "Get started",
+            "subTarget": "Welcome",
+            "style": "link"
+          },
+          {
+            "id": "96d6402c-ff6c-4cf8-961d-8fbe5c98be60",
```
[the remark this answers is not in the document]
Comment (arthurclares) [reply Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook:31]: Yes, we are working on that. Our plan is to have this completed in the next days.

File: gallery/workbook/Azure Advisor.json [hunk 5]
```diff
@@ -22,7 +22,7 @@
 					"id": "Workbooks/Azure Advisor/Cost Optimization",
 					"author": "Microsoft",
 					"name": "Cost Optimization (Preview)",
-					"description": "This workbook surfaces key recommendations that can directly impact cost savings. These recommendations include identifying and addressing idle resources, leveraging Azure Advisor recommendations such as Reserved Instances and Savings Plans for Compute, and optimizing the usage of oversized VMs. ",
+					"description": "Identify cost optimization opportunities for you Azure resources",
```
Comment (sebassem) [inline gallery/workbook/Azure Advisor.json:25]:
```suggestion
					"description": "Identify cost optimization opportunities for your Azure resources",
```

File: Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook (hunk 3 above)
Comment (arthurclares) [reply Workbooks/Azure Advisor/Cost Optimization/CostOptimization.workbook:237]: Let's wait for the UX review to make those changes at once.

**Comment by arthurclares** [comment]
Hey Parvathy, we have made the changes you suggested. Can you please review this PR again?

Let's know if there anything else you would like to change.

Thank you!

**Review by pkrishnanms — APPROVED** [review]

**Pull request merged**